TypeCobol.LanguageServer/WorkspaceProject.cs

```diff
@@ -0,0 +1,202 @@
+                if (!oldCopyFolders.Contains(fullPath))
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Configure this project
+        /// </summary>
+        /// <param name="format">The new Document format if any, null otherwise</param>
+        /// <param name="options">The new options if any, null otherwise</param>
+        /// <param name="analyzerProviderWrapper">The new Analyzer provider if any, null otherwise</param>
+        /// <param name="copyFolders">The new list of copy folders associated to this project if any, null otherwise</param>
+        public void Configure(DocumentFormat format, TypeCobolOptions options, AnalyzerProviderWrapper analyzerProviderWrapper, List<string> copyFolders)
+        {
+            bool updateFormat = false;
+            var newFormat = Project.Format;
+            if (format != null)
+            {
+                updateFormat = true;
+                newFormat = format;
+            }
+
+            bool updateOptions = false;
+            var newOptions = Project.CompilationOptions;
+            if (options != null)
+            {
+                updateOptions = true;
+                newOptions = options;
+            }
+
+            bool updateAnalyzerProvider = false;
+            var newAnalyzerProvider = Project.AnalyzerProvider;
+            if (analyzerProviderWrapper != null)
+            {
+                updateAnalyzerProvider = true;
+                newAnalyzerProvider = analyzerProviderWrapper;
+            }
+
+            var oldCopyFolders = GetCopyFolders();
+            bool updateCopyFolders = false;
+            var newCopyFolders = oldCopyFolders.ToList();
+            if (copyFolders != null && AreCopyFoldersSame(oldCopyFolders, copyFolders))
+            {
+                updateCopyFolders = true;
+                newCopyFolders = copyFolders;
+            }
+
+            if (updateFormat || updateOptions || updateAnalyzerProvider || updateCopyFolders)
+            {
+                var oldCompilationProject = Project;
+                var newCompilationProject = new CompilationProject(oldCompilationProject.Name, oldCompilationProject.RootDirectory, Helpers.DEFAULT_EXTENSIONS, newFormat, newOptions, newAnalyzerProvider);
+                foreach (var newCopyFolder in newCopyFolders)
+                {
+                    newCompilationProject.SourceFileProvider.AddLocalDirectoryLibrary(newCopyFolder, false, Helpers.DEFAULT_COPY_EXTENSIONS, newFormat);
+                }
+
+                Project = newCompilationProject;
+                // Change the target CompilationProject instance for each document.
+                foreach (var docCtx in _openedDocuments.Values)
+                {
+                    docCtx.FileCompiler.CompilationProject = Project;
+                }
+                _workspace.ScheduleRefresh(this, updateCopyFolders);
```

Comment:
In fact, since we have a new instance of CompilationProject, there is no need to clear the cache.

Reply:
Changed in [8d9c2da](https://github.com/TypeCobolTeam/TypeCobol/pull/2158/commits/8d9c2daba074a18dbf5260d351d3278d470ebf44)